de.dashup.shared/src/main/java/de/dashup/shared/Section.java

```diff
@@ -41,12 +41,20 @@ public String getName() {
         return widgets;
     }
 
-    public int getPredecessor() {
-        return predecessorID;
+    public int getIndex() {
+        return index;
     }
 
     @Override
     public int getId() {
         return sectionID;
     }
+
+    @Override
+    public int compareTo(Section sectionToCompare) {
+        if(index > sectionToCompare.getIndex()){
+            return 1;
+        }
+        return -1;
```

Comment:
I think this could be simplified by `return ((Integer) this.getIndex()).compareTo(widgetToCompare.getIndex())`. Besides this method does not cover equality (although this shouldn't be possible)